### PR TITLE
chore: const fold function last_query_id's argument.

### PR DESCRIPTION
**Pull request by RinChanNOWWW**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Const fold function last_query_id's argument.

### CODE REVIEW

File: src/query/sql/src/planner/semantic/type_check.rs [hunk 1]
```diff
@@ -1646,57 +1647,50 @@ impl<'a> TypeChecker<'a> {
...
-                        } => {
-                            if let Expr::Literal { span: _, lit: Literal::UInt64(l) } = **left
-                                && let Expr::Literal { span: _, lit: Literal::UInt64(r) } = **right {
-                                match op {
-                                    BinaryOperator::Plus => (l + r) as i32,
-                                    BinaryOperator::Minus => (l - r) as i32,
-                                    _ => -1,
-                                }
-                            } else { -1 }
+                let res: Result<i64> = try {
```
[the remark this answers is not in the document]
Comment (sundy-li) [reply src/query/sql/src/planner/semantic/type_check.rs:1650]: eg: 

```
fn get_number_arg<T: Num>() -> T
```

File: src/query/sql/src/planner/semantic/type_check.rs [hunk 2]
```diff
@@ -1646,57 +1647,50 @@ impl<'a> TypeChecker<'a> {
...
+                    if args.len() != 1 {
```
Comment (ariesdevil) [inline src/query/sql/src/planner/semantic/type_check.rs:1651]: args can be 0 or 1, if no argument supplied, index should be -1

File: src/query/expression/src/type_check.rs [hunk 3]
```diff
@@ -228,6 +231,34 @@ fn wrap_nullable_for_try_cast(span: Span, ty: &DataType) -> Result<DataType> {
     }
 }
 
+pub fn check_number<Index: ColumnIndex, T: Number>(
+    span: Span,
+    func_ctx: FunctionContext,
+    expr: &Expr<Index>,
+    fn_registry: &FunctionRegistry,
+) -> Result<T> {
+    let (expr, _) = ConstantFolder::fold(expr, func_ctx, fn_registry);
```
Comment (sundy-li) [inline src/query/expression/src/type_check.rs:240]: just use evaluator and 'to_xxx'

Comment (sundy-li) [reply src/query/expression/src/type_check.rs:240]: Or a CastExpr to `NumberType::<T>::data_type()`

Comment (RinChanNOWWW) [reply src/query/expression/src/type_check.rs:240]: This needs to new a `Evaluator` and construct a `DataBlock`. I think using `ConstantFolder` is more convenient.

**Review by sundy-li — APPROVED** [review]

**Review by ariesdevil — APPROVED** [review]

**Pull request merged**